source/icu.net/NativeMethods/NativeMethods.cs

```diff
@@ -176,6 +176,12 @@ internal static string DirectoryOfThisAssembly
 #endif
 #if NET
 				var managedPath = currentAssembly.Location;
+				// If the application is published as a single file, Assembly.Location will be null.
+				// Per the warning https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/warnings/il3000 we should use AppContext.BaseDirectory instead.
+				if (string.IsNullOrEmpty(managedPath))
+				{
+					managedPath = AppContext.BaseDirectory;
+				}
```

Comment:
Could probably also do it similar to the `#else`:

```suggestion
				// If the application is published as a single file, Assembly.Location will be null.
				// Per the warning https://learn.microsoft.com/en-us/dotnet/core/deploying/single-file/warnings/il3000 we should use AppContext.BaseDirectory instead.
				var managedPath = currentAssembly.Location ?? AppContext.BaseDirectory;
```

Reply:
Actually Location will be empty not null, I should update the comment.

Reply:
So `var managedPath = currentAssembly.Location || AppContext.BaseDirectory;` would work then?

Reply:
No, because that would always return `currentAssembly.Location` (because it returns `string.Empty` instead of `null`).

Reply:
Oops, sorry, I spend more time with JS where the empty string is falsy.